datasets/bprec/bprec.py

```diff
@@ -0,0 +1,168 @@
+                            "to": datasets.Value("int32"),
+                            "type": datasets.features.ClassLabel(
+                                names=[
+                                    "PRODUCT_NAME",
+                                    "PRODUCT_NAME_IMP",
+                                    "PRODUCT_NO_BRAND",
+                                    "BRAND_NAME",
+                                    "BRAND_NAME_IMP",
+                                    "VERSION",
+                                    "PRODUCT_ADJ",
+                                    "BRAND_ADJ",
+                                    "LOCATION",
+                                    "LOCATION_IMP",
+                                ]
+                            ),
+                        },
+                    }
+                ),
+            }
+        )
+        return datasets.DatasetInfo(
+            # This is the description that will appear on the datasets page.
+            description=_DESCRIPTION,
+            # This defines the different columns of the dataset and their types
+            features=features,  # Here we define them above because they are different between the two configurations
+            # If there's a common (input, target) tuple from the features,
+            # specify them here. They'll be used if as_supervised=True in
+            # builder.as_dataset.
+            supervised_keys=None,
+            # Homepage of the dataset for documentation
+            homepage=_HOMEPAGE,
+            # License for the dataset if available
+            license=_LICENSE,
+            # Citation for the dataset
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        # TODO: This method is tasked with downloading/extracting the data and defining the splits depending on the configuration
+        # If several configurations are possible (listed in BUILDER_CONFIGS), the configuration selected by the user is in self.config.name
+
+        # dl_manager is a datasets.download.DownloadManager that can be used to download and extract URLs
+        # It can accept any type or nested list/dict and will give back the same structure with the url replaced with path to local files.
+        # By default the archives will be extracted and a path to a cached folder where they are extracted is returned instead of the archive
+        downloaded_files = dl_manager.download_and_extract(_URLs)
+        return [
+            datasets.SplitGenerator(name="tele", gen_kwargs={"filedirs": [downloaded_files["tele"]]}),
+            datasets.SplitGenerator(name="electro", gen_kwargs={"filedirs": [downloaded_files["electro"]]}),
+            datasets.SplitGenerator(name="cosmetics", gen_kwargs={"filedirs": [downloaded_files["cosmetics"]]}),
+            datasets.SplitGenerator(name="banking", gen_kwargs={"filedirs": [downloaded_files["banking"]]}),
```

Comment:
Can you use configurations to differentiate between all those topics instead of splits ?
Splits are usually used to separate the chunks of data for the different steps of the evaluation of models

I also like the idea of having a config `all`. If you think it makes sense you can also add it.

Reply:
Thanks for the feedback! Yes, I'll change it to configuration, hope to be able to do it by tomorrow morning :)